src/Compiler/Facilities/DiagnosticsLogger.fs

```diff
@@ -920,16 +909,56 @@ type StackGuard(maxDepth: int, name: string) =
     static member GetDepthOption(name: string) =
         GetEnvInteger ("FSHARP_" + name + "StackGuardDepth") StackGuard.DefaultDepth
 
-type CaptureDiagnosticsConcurrently() =
-    let target = DiagnosticsThreadStatics.DiagnosticsLogger
-    let loggers = ResizeArray()
+module MultipleDiagnosticsLoggers =
```

Comment:
This code looks a bit subtle, do you think we can add some explicit testing for this?

Reply:
Right. Specifically as for now there is not test checking that the diags are committed in the right order and timely.

Reply:
added a test.

Reply:
Thansk!